ocrd_tesserocr/recognize.py

```diff
@@ -207,12 +207,15 @@ def process(self):
         if 'model' in self.parameter:
             model = self.parameter['model']
             for sub_model in model.split('+'):
+                # XXX this will ensure that the models are downloaded if not already in MODEL_LOCATION
+                # TODO handle existing/missing .traineddata extension
+                self.resolve_resource(sub_model)
```

Comment:
This must be adapted once #167 is merged.

Reply:
Yes. But I wonder how we can ensure this still only uses our single unique resource location (for which we even have an envvar override). `Processor.resolve_resource` would not only mix in other locations, it could even download to other locations. None of these other locations would actually work, however. So perhaps we should re-implement that function here. 

Or we drop it and just keep raising an exception if any model is not in `get_languages()`?

Also, perhaps we should move the model initialization code from `process` to some `setup` which we call from the constructor if a workspace was passed in. (So a future processor server or the current workflow server could pick it up.) See ocrd_cis for that pattern.

Reply:
I think you're right, that we shouldn't be too clever for our own good here. Download-on-demand is nice and all but even more so is clarity. For the CI e.g. I resorted to just doing `ocrd resmgr download` explicitly instead of relying on this mechanism. So I'll just remove it, better to have a clear error message than a potentially buggy on-demand downloading.

Reply:
44dd99d

Reply:
> Also, perhaps we should move the model initialization code from process to some setup which we call from the constructor if a workspace was passed in. (So a future processor server or the current workflow server could pick it up.) See ocrd_cis for that pattern.

Valid point but let's do this in a separate PR.

Reply:
> So I'll just remove it, better to have a clear error message than a potentially buggy on-demand downloading.

But then we need to make an exception in our documentation. So how about the other processors – do we really want automatic downloading of resources in a processor, now that your `ocrd resmgr` is super mighty?

(One can argue against that as a general convenience feature, by remarking that it is not guaranteed to work on all requested/configured resources anyway, and that workflow flaws need to be detected early-on independently. Especially if we want to separate setup and processing.)

Reply:
The main argument against on-demand downloading IMHO is that it hides errors. I am right now debugging the resource-manager adaption in ocrd_anybaseocr and it doesn't find the models and tries to download them again, which is actually broken right now, as I just learned 😠 And you wouldn't get a nice progress bar or anything, it would just appear to hang while downloading gigabytes of redundant data.... So, I'll patch this feature out of the code and documentation, best to use `ocrd resmgr` exclusively for this.